File: dotWasmBenchmark.DotNet/Calc.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace BakkBenchmark.DotNet
{
    public static class Calc
    {
        public static StatResult Calculate(double[] list)
        {
            var q1 = list[list.Length / 4];
            var median = list[list.Length / 2];
            var q3 = list[list.Length / 4 * 3];

            var avg = list.Average();
            var stDevSum = list.Sum(d => (d - avg) * (d - avg));
            var stDev = Math.Sqrt(stDevSum / list.Length);

            return new StatResult() {
                Q1 = q1,
                Median = median,
                Q3 = q3,
                Avg = avg,
                StDev = stDev
            };
        }
    }

    public struct StatResult {
        public double Q1 { get; set; }
        public double Median { get; set; }
        public double Q3 { get; set; }
        public double Avg { get; set; }
        public double StDev { get; set; }
    }
}

File: dotWasmBenchmark.JavaScript/src/Calc.ts
class Calc {
    public static calculate(list: number[]): any {
        let sum = 0.0;

        for (let i = 0; i < list.length; i++) {
            sum += list[i];
        }

        const avg = sum / list.length;

        let stDevSum = 0.0;

        for (let i = 0; i < list.length; i++) {
            stDevSum += (list[i] - avg) * (list[i] - avg);
        }

        const stDev = Math.sqrt(stDevSum / list.length);

        return {
            q1: list[list.length / 4],
            median: list[list.length / 2],
            q3: list[list.length / 4 * 3],
            avg: avg,
            stDev: stDev
        };
    }
}
